File: Soal2.cpp
/*
 Nama : Wilson Angelie Tan, Stan F., Theophilus Samuel G.
 NPM :140810230024, 140810230046, 140810230054
 Tanggal Buat :13/10/2024
 Deskripsi :penjumlahan matriks menggunakan array dua dimensi dalam bentuk OOP2 class bahasa C++
*/

#include <iostream>
using namespace std;

class Matriks {
private:
    int baris;
    int kolom;
    int nilai[100][100];

public:
    Matriks() : baris(0), kolom(0) {}

    void setBaris(int baris) {
        this->baris = baris;
    }

    void setKolom(int kolom) {
        this->kolom = kolom;
    }

    void inputMatriks() {
        cout << "Masukkan Besar baris matriks: ";
        cin >> this->baris;
        cout << "Masukkan Besar kolom matriks: ";
        cin >> this->kolom;

        for (int i = 0; i < this->baris; i++) {
            for (int j = 0; j < this->kolom; j++) {
                cout << "Masukkan Matriks baris ke-" << (i + 1) << " kolom ke-" << (j + 1) << ": ";
                cin >> this->nilai[i][j];
            }
        }
    }

    void outputMatriks() {
        if (this->baris == 0 || this->kolom == 0) {
            cout << "Matriks belum diinput!!" << endl;
            return;
        }
        for (int i = 0; i < this->baris; i++) {
            for (int j = 0; j < this->kolom; j++) {
                cout << this->nilai[i][j] << " ";
            }
            cout << endl;
        }
    }

    void penjumlahanVoid(const Matriks& m1, const Matriks& m2) {
        this->setBaris(m1.baris);
        this->setKolom(m1.kolom);
        if (m1.baris != m2.baris || m1.kolom != m2.kolom) {
            cout << "Ukuran Ordo Matriks 1 dan 2 tidak sama, Oleh karena itu Penjumlahan tidak dapat dilakukan!" << endl;
            return;
        }
        for (int i = 0; i < this->baris; i++) {
            for (int j = 0; j < this->kolom; j++) {
                this->nilai[i][j] = m1.nilai[i][j] + m2.nilai[i][j];
            }
        }
    }

    Matriks penjumlahanFungsi(const Matriks& m2) {
        Matriks m3;
        m3.setBaris(this->baris);
        m3.setKolom(this->kolom);
        if (this->baris != m2.baris || this->kolom != m2.kolom) {
            cout << "Ukuran Ordo Matriks 1 dan 2 tidak sama, Oleh karena itu Penjumlahan tidak dapat dilakukan!" << endl;
            return m3;
        }
        for (int i = 0; i < this->baris; i++) {
            for (int j = 0; j < this->kolom; j++) {
                m3.nilai[i][j] = this->nilai[i][j] + m2.nilai[i][j];
            }
        }
        return m3;
    }
};

void menu(Matriks& m1, Matriks& m2) {
    int pilihan = 1;
    while (pilihan != 0) {
        cout << "\n\tMenu" << endl;
        cout << "===============" << endl;
        cout << "1. Input Matriks 1" << endl;
        cout << "2. Input Matriks 2" << endl;
        cout << "3. Output Matriks 1" << endl;
        cout << "4. Output Matriks 2" << endl;
        cout << "5. VOID Penjumlahan Matriks 1 dan 2" << endl;
        cout << "6. FUNGSI Penjumlahan Matriks 1 dan 2" << endl;
        cout << "0. Exit Program!!" << endl;
        cout << "> ";
        cin >> pilihan;

        switch (pilihan) {
            case 1:
                m1.inputMatriks();
                break;
            case 2:
                m2.inputMatriks();
                break;
            case 3:
                m1.outputMatriks();
                break;
            case 4:
                m2.outputMatriks();
                break;
            case 5: {
                Matriks m3;
                m3.penjumlahanVoid(m1, m2);
                m3.outputMatriks();
                break;
            }
            case 6: {
                Matriks m4 = m1.penjumlahanFungsi(m2);
                m4.outputMatriks();
                break;
            }
            case 0:
                cout << "Exit Program!!" << endl;
                break;
            default:
                cout << "Pilihan tidak ada pada menu!!" << endl;
                break;
        }
    }
}

int main() {
    Matriks m1, m2;
    menu(m1, m2);
    return 0;
}

File: Soal2.py
class Matriks:
    def __init__(self, baris=0, kolom=0, nilai=None):
        self.__baris = baris
        self.__kolom = kolom
        if nilai is None:
            self.__nilai = [[0 for _ in range(self.__kolom)] for _ in range(self.__baris)]
        else:
            self.__nilai = nilai
    
    def setBaris(self, baris):
        self.__baris = baris
        self.__nilai = [[0 for _ in range(self.__kolom)] for _ in range(self.__baris)]
        
    def setKolom(self, kolom):
        self.__kolom = kolom
        self.__nilai = [[0 for _ in range(self.__kolom)] for _ in range(self.__baris)]
        
    def setNilai(self, nilai):
        self.__nilai = nilai
    
    def inputBaris(self):
        self.__baris = int(input("Masukkan Besar baris matriks : "))
        self.__nilai = [[0 for _ in range(self.__kolom)] for _ in range(self.__baris)]
        
    def inputKolom(self):
        self.__kolom = int(input("Masukkan Besar kolom matriks : "))
        self.__nilai = [[0 for _ in range(self.__kolom)] for _ in range(self.__baris)]
        
    def inputBK(self):
        self.inputBaris()
        self.inputKolom()
        
    def inputMatriks(self):
        self.inputBK()
        
        for a in range(self.__baris):
            for b in range(self.__kolom):
                self.__nilai[a][b] = int(input(f"Masukkan Matriks baris ke-{a+1} kolom ke-{b+1} : "))
                
    def outputMatriks(self):
        if(self.__baris == 0 | self.__kolom == 0):
            print("Matriks belum diinput!!")
            return
        for a in range(self.__baris):
            for b in range(self.__kolom):
                print(self.__nilai[a][b], " ", end="")
            print("")

    def penjumlahanVoid(self, m1, m2):
        self.setBaris(m1.__baris)
        self.setKolom(m1.__kolom)
        
        if(m1.__baris != m2.__baris or m1.__kolom != m2.__kolom):
            print("Ukuran Ordo Matriks 1 dan 2 tidak sama, Oleh karena itu Penjumlahan tidak dapat dilakukan!")
            return
        
        for a in range(self.__baris):
            for b in range(self.__kolom):
                self.__nilai[a][b] = m1.__nilai[a][b] + m2.__nilai[a][b]
                
    def penjumlahanFungsi(self, m2):
        m3 = Matriks()
        m3.setBaris(self.__baris)
        m3.setKolom(self.__kolom)
        
        if(self.__baris != m2.__baris or self.__kolom != m2.__kolom):
            print("Ukuran Ordo Matriks 1 dan 2 tidak sama, Oleh karena itu Penjumlahan tidak dapat dilakukan!")
            return m3
        
        for a in range(self.__baris):
            for b in range(self.__kolom):
                m3.__nilai[a][b] = self.__nilai[a][b] + m2.__nilai[a][b]
                
        return m3
    
def menu(m1, m2):
    pilihan = 1
    
    while(pilihan != 0):
        print("\n\tMenu")
        print("===============")
        print("1. Input Matriks 1")
        print("2. Input Matriks 2")
        print("3. Output Matriks 1")
        print("4. Output Matriks 2")
        print("5. VOID Penjumlahan Matriks 1 dan 2")
        print("6. FUNGSI Penjumlahan Matriks 1 dan 2")
        print("0. Exit Program!!")
        pilihan = int(input("> "))
        
        if(pilihan == 1):
            m1.inputMatriks()
        elif(pilihan == 2):
            m2.inputMatriks()
        elif(pilihan == 3):
            m1.outputMatriks()
        elif(pilihan == 4):
            m2.outputMatriks()
        elif(pilihan == 5):
            m3 = Matriks()
            m3.penjumlahanVoid(m1, m2)
            m3.outputMatriks()
        elif(pilihan == 6):
            m4 = m1.penjumlahanFungsi(m2)
            m4.outputMatriks()
        elif(pilihan == 0):
            print("Exit Program!!")
        else:
            print("Pilihan tidak ada pada menu!!")

m1 = Matriks()
m2 = Matriks()
menu(m1, m2)